packages/astro/test/redirects.test.js

```diff
@@ -345,4 +345,22 @@ describe('Astro.redirect', () => {
 			assert.equal(secretHtml.includes('url=/login'), true);
 		});
 	});
+
+	describe('should fail for redirects that cannot be mapped ', () => {
+		it('should fail for redirects that cannot be mapped', async () => {
+			fixture = await loadFixture({
+				root: './fixtures/redirects/',
+				output: 'static',
+				redirects: {
+					'/old/[category]/': '/old/[category]/[slug]',
+				},
+			});
+			try {
+				await fixture.build();
+				assert.fail('Expected build to fail');
+			} catch {
+				assert.ok(true);
+			}
```

Comment:
This test passes without your change, because the build is failing anyway. Can you assert that the right error is being thrown? I just checked and it seems it's still throwing the `GetStaticPathsRequired` here.